provider/pkg/provider/helm_release.go

```diff
@@ -1441,6 +1455,54 @@ func mapToInterface(in any) any {
 	return in
 }
 
+// searchProgramDirectory implements a best-effort search for a chart in the program directory.
+func searchProgramDirectory(name, version string) (*string, *helmchart.Metadata, bool) {
```

Comment:
Can you explain the ordering here?

It goes (as implemented):
- `NAME.tgz`
- `NAME-VERSION.tgz`
- `NAME` (dir)
For example, why isn't there a `NAM-VERSION` (dir) option?

Unit tests for this function and `searchHelmRepositories` would be appreciated. It's not clear to me how the go test you added exercises this functionality.

Reply:
I hadn't considered a `NAME-VERSION` directory because I've not encountered it before; the other options stem from ordinary Helm usage, e.g. saving or extracting a chart archive. It is a reasonable extension, I'll try it.

The go test named `Helm Release Import (Tool)` is exercising each of these options by selecting a directory in the initial step, e.g. `step1-local-tar`. As for how it is asserted, the import step fails if the synthesized inputs don't match the program inputs.